src/test/java/org/zalando/nakadi/service/timeline/TimelineServiceTest.java

```diff
@@ -0,0 +1,180 @@
+package org.zalando.nakadi.service.timeline;
+
+import com.google.common.collect.ImmutableList;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Matchers;
+import org.mockito.Mockito;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.support.TransactionTemplate;
+import org.zalando.nakadi.config.NakadiSettings;
+import org.zalando.nakadi.config.SecuritySettings;
+import org.zalando.nakadi.domain.EventType;
+import org.zalando.nakadi.domain.Storage;
+import org.zalando.nakadi.domain.Timeline;
+import org.zalando.nakadi.exceptions.InternalNakadiException;
+import org.zalando.nakadi.exceptions.NoSuchEventTypeException;
+import org.zalando.nakadi.exceptions.NotFoundException;
+import org.zalando.nakadi.exceptions.TimelineException;
+import org.zalando.nakadi.exceptions.UnableProcessException;
+import org.zalando.nakadi.repository.TopicRepository;
+import org.zalando.nakadi.repository.TopicRepositoryHolder;
+import org.zalando.nakadi.repository.db.EventTypeCache;
+import org.zalando.nakadi.repository.db.StorageDbRepository;
+import org.zalando.nakadi.repository.db.TimelineDbRepository;
+import org.zalando.nakadi.security.FullAccessClient;
+import org.zalando.nakadi.util.UUIDGenerator;
+import org.zalando.nakadi.utils.EventTypeTestBuilder;
+
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+public class TimelineServiceTest {
+
+    private final SecuritySettings securitySettings = Mockito.mock(SecuritySettings.class);
+    private final EventTypeCache eventTypeCache = Mockito.mock(EventTypeCache.class);
+    private final StorageDbRepository storageDbRepository = Mockito.mock(StorageDbRepository.class);
+    private final TimelineSync timelineSync = Mockito.mock(TimelineSync.class);
+    private final NakadiSettings nakadiSettings = Mockito.mock(NakadiSettings.class);
+    private final TimelineDbRepository timelineDbRepository = Mockito.mock(TimelineDbRepository.class);
+    private final TopicRepositoryHolder topicRepositoryHolder = Mockito.mock(TopicRepositoryHolder.class);
+    private final TransactionTemplate transactionTemplate =
+            new TransactionTemplate(Mockito.mock(PlatformTransactionManager.class));
+
+    private final TimelineService timelineService = new TimelineService(securitySettings, eventTypeCache,
+            storageDbRepository, timelineSync, nakadiSettings, timelineDbRepository, topicRepositoryHolder,
+            transactionTemplate, new UUIDGenerator());
+
+    @Test
+    public void testCreateTimelineFromFake() throws Exception {
```

Comment:
This test has too many mocks. I think it would be better just to test it in the "acceptance tests".